gitlabform/gitlabform/processors/project/schedules_processor.py

```diff
@@ -0,0 +1,127 @@
+                        updated_schedule.get("id"),
+                        configuration.get(
+                            "schedules|" + schedule_description + "|variables"
+                        ),
+                    )
+                elif schedule_ids:
+                    logging.debug(
+                        "Replacing existing pipeline schedules '%s'",
+                        schedule_description,
+                    )
+                    for schedule_id in schedule_ids:
+                        self.gitlab.delete_pipeline_schedule(
+                            project_and_group, schedule_id
+                        )
+                    self.create_schedule_with_variables(
+                        configuration, project_and_group, schedule_description
+                    )
+                else:
+                    logging.debug(
+                        "Creating pipeline schedule '%s'", schedule_description
+                    )
+                    self.create_schedule_with_variables(
+                        configuration, project_and_group, schedule_description
+                    )
+
+    def create_schedule_with_variables(
+        self, configuration, project_and_group, schedule_description
+    ):
+        data = configuration.get("schedules|" + schedule_description)
+        created_schedule = self.gitlab.create_pipeline_schedule(
+            project_and_group,
+            schedule_description,
+            data.get("ref"),
+            data.get("cron"),
+            optional_data=data,
+        )
+        self.__set_schedule_variables(
+            project_and_group,
+            created_schedule.get("id"),
+            configuration.get("schedules|" + schedule_description + "|variables"),
+        )
+
+    def __set_schedule_variables(self, project_and_group, schedule_id, variables):
+        schedule = self.gitlab.get_pipeline_schedule(project_and_group, schedule_id)
+
+        existing_variables = schedule.get("variables")
+        if existing_variables:
+            logging.debug(
+                "Deleting variables for pipeline schedule '%s'", schedule["description"]
+            )
+
+            for variable in existing_variables:
+                self.gitlab.delete_pipeline_schedule_variable(
+                    project_and_group, schedule_id, variable.get("key")
+                )
+
+        for variable_key, variable_data in variables.items():
+            self.gitlab.create_pipeline_schedule_variable(
+                project_and_group,
+                schedule_id,
+                variable_key,
+                variable_data.get("value"),
+                variable_data,
+            )
+
+    @staticmethod
+    def __group_schedule_ids_by_description(schedules) -> Dict[str, List[str]]:
+        schedule_ids_by_description: Dict[str, List[str]] = {}
+
+        for schedule in schedules:
+            description = schedule["description"]
+            if description in schedule_ids_by_description:
```

Comment:
Nit: you can set dict's default value for non-existent key to an empty list `[]` to avoid this `if`. See https://github.com/egnyte/gitlabform/blob/master/gitlabform/gitlabform/processors/group/group_members_processor.py#L28-28 .

Reply:
Ah, that's a method I missed in Python quite a few times :) I've adjusted the code.